### ISSUE TITLE
`macho/libunwind.sh` test failure on M1 macOS Monterey (mold 1.5.0)

### ISSUE THREAD

**Issue by carlocab** [rui314/mold#735]
I'm trying to update Homebrew's Mold package to 1.5.0 at Homebrew/homebrew-core#111776.

I'm struggling with a test failure for `macho/libunwind.sh`, which errors with:
```
  Testing libunwind ... mold: error: undefined symbol: out/test/macho/arm64/libunwind/a.o: _objc_msgSend$raise:format:
```
https://github.com/Homebrew/homebrew-core/actions/runs/3133481044/jobs/5087212968#step:6:60

This occurs only on macOS 12 running on ARM. In particular, this doesn't seem to be happening on any version of macOS we test on Intel (10.15, 11, 12), or on macOS 11 on ARM. This same test succeeded in the previous version of Mold (1.4.2).

I suspect this has something to do with Xcode 14, since I am unable to reproduce this issue locally and the only difference between my setup and the one on Homebrew CI is that I am using a slightly older toolchain (CLT for Xcode 13.4).

Any assistance would be appreciated.

**Comment by carlocab** [comment]
Oh, actually, `macho/objc-selector.sh` fails too, with the same error. I'm going to guess that `macho/objc.sh` will perform similarly.

To be clear, Xcode 14 alone is not sufficient to produce this failure, as we are running Xcode 14 on Intel macOS 12 too. It seems you need to be on an M1 Mac running Monterey with Xcode 14.

**Comment by ZhongRuoyu** [comment]
> I suspect this has something to do with Xcode 14

Yeah, that's related to llvm/llvm-project#56034. Also reported here at #758.
